despatch_advice_import/wizard/despatch_advice_import.py

```diff
@@ -0,0 +1,327 @@
+
+        qty = line["qty"]
+        backorder_qty = line["backorder_qty"]
+        moves_qty = sum(moves.mapped("product_qty"))
+
+        if float_compare(qty, moves_qty, precision_digits=precision) >= 0:
+            return
+
+        # confirmed qty < ordered qty
+        move_ids_to_backorder = []
+        move_ids_to_cancel = []
+        for move in moves:
+            self._check_picking_status(move.picking_id)
+            if (
+                float_compare(qty, move.product_uom_qty, precision_digits=precision)
+                >= 0
+            ):
+                # qty planned => qty into the stock move: Keep it
+                qty -= move.product_uom_qty
+                continue
+            if (
+                qty
+                and float_compare(qty, move.product_uom_qty, precision_digits=precision)
+                < 0
+            ):
+                # qty planned < qty into the stock move: Split it
+                new_vals = move._split(move.product_uom_qty - qty)
+                move.quantity_done = move.product_qty
+                move._action_done()
+                move = self.env["stock.move"].create(new_vals[0])
+                move._action_confirm()
+
+            qty -= move.product_uom_qty
+            if not backorder_qty:
+                # if no backorder -> we must cancel the move
+                move_ids_to_cancel.append(move.id)
+                continue
+            # from here we process the backorder qty
+            # we distribute this qty into the remaining moves and
+            # if this qty is < than the expected one, we split and cancel the
+            # remaining qty
+            if (
+                float_compare(
+                    backorder_qty, move.product_uom_qty, precision_digits=precision
+                )
+                < 0
+            ):
+                # backorder_qty < qty into the move -> split the move
+                # anf cancel remaining qty
+                move._action_confirm(merge=False)
+                new_vals = move._split(move.product_uom_qty - backorder_qty)
+                move_ids_to_cancel.append(self.env["stock.move"].create(new_vals[0]).id)
+
+            backorder_qty -= move.product_uom_qty
+            move_ids_to_backorder.append(move.id)
+
+        # move backorder moves to a backorder
+        if move_ids_to_backorder:
+            moves_to_backorder = self.env["stock.move"].browse(move_ids_to_backorder)
+            self._add_moves_to_backorder(moves_to_backorder)
+        # cancel moves to cancel
+        if move_ids_to_cancel:
+            moves_to_cancel = self.env["stock.move"].browse(move_ids_to_cancel)
+            moves_to_cancel._action_cancel()
+
+    def _add_moves_to_backorder(self, moves):
+        """
+        Add the move the picking's backorder
+        return the backorder associated to the current picking. If no backorder
+        exists, create a new one.
```

Comment:
```suggestion
        """Add the move the picking's backorder.
        
        If no backorder exists, create a new one.
        
        :return: backorder associated to the current picking. 
```
However, it does not return any record....